### PR TITLE
Abp React Implementation

### PR DESCRIPTION
**Pull request by ryoldash**
Initial pull request. Code refactoring will be done when required.

### CODE REVIEW

File: reactjs/src/components/Layout/UserLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import DocumentTitle from 'react-document-title';
+import { Switch, Route, Redirect } from 'react-router-dom';
+import utils from 'src/utils/utils';
+import { userRouter } from '../Router/router.config';
+import { Col } from 'antd';
+// import "./UserLayout.css";
```
Comment (alirizaadiyahsi) [inline reactjs/src/components/Layout/UserLayout.tsx]: Unused comment line

File: reactjs/src/components/Router/ProtectedRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { isGranted } from 'src/lib/abpUtility';
+// import { Alert } from 'antd';
+// import { isGranted } from 'src/lib/abpUtility';
```
Comment (alirizaadiyahsi) [inline reactjs/src/components/Router/ProtectedRoute.tsx]: Unused comment lines

File: reactjs/src/scenes/Login/index.validation.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// const rules = [];
+
+// export default rules;
```
Comment (alirizaadiyahsi) [inline reactjs/src/scenes/Login/index.validation.ts]: rules are defined in `scenes/Login/index.tsx`, so no need to keep this file.

File: reactjs/src/services/role/dto/getAllPermissionsOutput.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    export interface GetAllPermissionsOutput {
+        name: string;
+        displayName: string;
+        description: string;
+        id: number;
+    }
+
+   
+
+
```
Comment (alirizaadiyahsi) [inline reactjs/src/services/role/dto/getAllPermissionsOutput.ts]: There are unnecessary empty lines and whitespaces in files. I don't write review to all files, but you can check all of them, and maybe you can prefer to use a linter.

File: reactjs/src/stores/tenantStore.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+  @action
+  async create(createTenantInput: CreateTenantInput) {
+  
+    var result = await tenantService.create(createTenantInput);
+    console.log(result);
+    
+    // this.tenants.items.unshift(result);
+   
+    //
```
Comment (alirizaadiyahsi) [inline reactjs/src/stores/tenantStore.ts]: Remove commented out lines and all `console.log` lines in all JS files.

File: reactjs/src/stores/tokenAuthStore.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import { AuthenticationResultModel } from './../services/tokenAuth/dto/authenticationResultModel';
+
+const tokenStorageName = 'abp.AuthToken';
+
+class TokenAuthStore {
+  @observable authenticateResult: AuthenticationResultModel | null;
+
+  constructor() {
+    //Kullanıcı herhangi bir nedende dolayı sayfadan cıkışşsa
+    //storagedan tokenı tekrar stora almayı sağlar
```
Comment (alirizaadiyahsi) [inline reactjs/src/stores/tokenAuthStore.ts]: Could you write comments in English? And I think there are unnecessary comments. For example: `// clear local storage`, ` //false ise login olunmamıştır` and etc. Could you please remove these unnecessary comments?

File: reactjs/src/stores/userStore.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+class UserStore {
+  @observable users: PagedResultDto<GetUserOutput>;
+  @observable editUser: CreateOrUpdateUserInput;
+  @observable roles: GetRoles[] = [];
+
+  @action
+  async create(createUserInput: CreateOrUpdateUserInput) {
+    
+    var result = await userService.create(createUserInput);
+    console.log(result);
```
Comment (alirizaadiyahsi) [inline reactjs/src/stores/userStore.ts]: Remove `console.log` lines.

**Comment by alirizaadiyahsi** [comment]
I run the application and there some console errors. These errors cause dashboard down. If I click users, console errors occur, when I return to dashboard it looks like following:

image

And ui is failing too (no padding/margin between cards and some alignment problems):

image

image

image

File: reactjs/src/components/Layout/UserLayout.tsx (hunk 1 above)
Comment (ryoldash) [reply reactjs/src/components/Layout/UserLayout.tsx]: Resolved

File: reactjs/src/scenes/Login/index.validation.ts (hunk 3 above)
Comment (ryoldash) [reply reactjs/src/scenes/Login/index.validation.ts]: Resolved

File: reactjs/src/services/role/dto/getAllPermissionsOutput.ts (hunk 4 above)
Comment (ryoldash) [reply reactjs/src/services/role/dto/getAllPermissionsOutput.ts]: Resolved

File: reactjs/src/stores/tenantStore.ts (hunk 5 above)
Comment (ryoldash) [reply reactjs/src/stores/tenantStore.ts]: Resolved

File: reactjs/src/stores/userStore.ts (hunk 7 above)
Comment (ryoldash) [reply reactjs/src/stores/userStore.ts]: Resolved

File: reactjs/src/stores/tokenAuthStore.ts (hunk 6 above)
Comment (ryoldash) [reply reactjs/src/stores/tokenAuthStore.ts]: Resolved

**Comment by alirizaadiyahsi** [comment]
Thanks @ryoldash. There is one thing, if I click users/roles and return back to dashboard, page looks like following : (I think some css code is not loaded)

image

**Comment by smtkby** [comment]
The problem will be solved as soon as possible.

**Comment by smtkby** [comment]
The problem has been fixed.

**Pull request merged**